redwood-treehouse-lazylayout-compose/src/jsMain/kotlin/app/cash/redwood/treehouse/lazylayout/compose/TreehouseLazyColumn.kt

```diff
@@ -24,49 +24,60 @@ import app.cash.redwood.treehouse.asZiplineTreehouseUi
 import app.cash.redwood.treehouse.lazylayout.api.LazyListIntervalContent
 
 @Composable
-fun DiffProducingWidget.Provider.LazyColumn(content: LazyListScope.() -> Unit) {
-  val lazyListScope = LazyListScope(this)
-  content(lazyListScope)
-  app.cash.redwood.treehouse.lazylayout.compose.LazyColumn(lazyListScope.intervals)
+public fun DiffProducingWidget.Provider.LazyColumn(content: LazyListScope.() -> Unit) {
+  val scope = TreehouseLazyListScope(this)
+  content(scope)
+  LazyColumn(scope.intervals)
 }
 
 @LayoutScopeMarker
-class LazyListScope(private val provider: DiffProducingWidget.Provider) {
-  private val _intervals = mutableListOf<LazyListIntervalContent>()
-  val intervals: List<LazyListIntervalContent> = _intervals
+public interface LazyListScope {
+  public fun items(
+    count: Int,
+    itemContent: @Composable (index: Int) -> Unit,
+  )
+}
+
+public inline fun <T> LazyListScope.items(
+  items: List<T>,
+  crossinline itemContent: @Composable (item: T) -> Unit,
+) {
+  items(items.size) {
+    itemContent(items[it])
+  }
+}
+
+private class TreehouseLazyListScope(private val provider: DiffProducingWidget.Provider) : LazyListScope {
+  val intervals = mutableListOf<LazyListIntervalContent>()
 
   private class Item(
     private val provider: DiffProducingWidget.Provider,
     private val content: @Composable (index: Int) -> Unit,
   ) : LazyListIntervalContent.Item {
 
     override fun get(index: Int): ZiplineTreehouseUi {
-      val treehouseUi = object : TreehouseUi {
-        @Composable
-        override fun Show() {
-          content(index)
-        }
-      }
+      val treehouseUi = IndexedTreehouseUi(content, index)
       return treehouseUi.asZiplineTreehouseUi(provider, widgetVersion = 1u)
     }
   }
 
-  fun items(
+  override fun items(
     count: Int,
     itemContent: @Composable (index: Int) -> Unit,
   ) {
-    _intervals += LazyListIntervalContent(
+    intervals += LazyListIntervalContent(
       count,
       itemProvider = Item(provider, itemContent),
     )
   }
 }
 
-inline fun <T> LazyListScope.items(
-  items: List<T>,
-  crossinline itemContent: @Composable (item: T) -> Unit,
-) = items(
-  count = items.size,
-) {
-  itemContent(items[it])
+private class IndexedTreehouseUi(
```

Comment:
Compose compiler was choking on this being an anonymous class inside a private class inside a private class so had to move.

Reply:
Filed as https://issuetracker.google.com/issues/264799026